# HG changeset patch
# User Ben Abbott <bpabbott@mac.com>
# Date 1244680390 14400
#      Wed Jun 10 20:33:10 2009 -0400
# Node ID 0307f5e5568c0714da288248d45fcad1975bebed
# Parent  e12a5ec6cf0d5fccc9e01944a928c639bdc66aba
__go_draw_axes__.m: Fix order when pushing group children onto kid list.

diff --git a/scripts/ChangeLog b/scripts/ChangeLog
--- a/scripts/ChangeLog
+++ b/scripts/ChangeLog
@@ -1,10 +1,13 @@
-2009-06-10  Marco Caliari  <marco.caliari@univr.it>
-
+2009-06-10  Marco Caliari <marco.caliari@univr.it>
+
+	* plot/hold.m: Add demo including a hggroup.
+	* plot/__go_draw_axes__.m: Fix order when pushing group children onto
+	the axes kid list.
 	* general/quadgk.m: Better waypoint transform.
 
 2009-06-09  David Bateman  <dbateman@free.fr>
 
 	* general/quadgk.m: Add test case and fixed doubly infinite 
 	waypoint transform for x = 0 case.
 
 2009-06-09  Marco Caliari <marco.caliari@univr.it>
diff --git a/scripts/plot/__go_draw_axes__.m b/scripts/plot/__go_draw_axes__.m
--- a/scripts/plot/__go_draw_axes__.m
+++ b/scripts/plot/__go_draw_axes__.m
@@ -1023,17 +1023,17 @@ function __go_draw_axes__ (h, plot_strea
 		     __do_enhanced_option__ (enhanced, obj), colorspec);
 	  endif
 
         case "hggroup"
 	  ## Push group children into the kid list.
 	  if (isempty (kids))
 	    kids = obj.children;
 	  elseif (! isempty (obj.children))
-	    kids = [obj.children; kids];
+	    kids = [kids; obj.children];
 	  endif
 
 	otherwise
 	  error ("__go_draw_axes__: unknown object class, %s",
 		 obj.type);
       endswitch
 
     endwhile
diff --git a/scripts/plot/hold.m b/scripts/plot/hold.m
--- a/scripts/plot/hold.m
+++ b/scripts/plot/hold.m
@@ -116,8 +116,23 @@ endfunction
 
 %!demo
 %! clf
 %! hold on
 %! plot (1:4, "-s")
 %! imagesc(1./hilb(4));
 %! hold off
 
+%!demo
+%! clf
+%! colormap (jet)
+%! t = linspace (-3, 3, 50);
+%! [x, y] = meshgrid (t, t);
+%! z = peaks (x, y);
+%! contourf (x, y, z, 10);
+%! hold ("on");
+%! plot (vec (x), vec (y), "^");
+%! patch ([-1.0 1.0 1.0 -1.0 -1.0], [-1.0 -1.0 1.0 1.0 -1.0], "red");
+%! xlim ([-2.0 2.0]);
+%! ylim ([-2.0 2.0]);
+%! colorbar ("SouthOutside");
+%! title ("Test script for some plot functions");
+
